components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -1,25 +1,238 @@
 #include "ekat/ekat_assert.hpp"
 #include "physics/rrtmgp/scream_rrtmgp_interface.hpp"
 #include "physics/rrtmgp/atmosphere_radiation.hpp"
+#include "physics/rrtmgp/rrtmgp_inputs_initializer.hpp"
+#include "mo_gas_concentrations.h"
 
 namespace scream {
-    RRTMGPRadiation::RRTMGPRadiation (const ekat::Comm& comm, const ekat::ParameterList& params) : m_rrtmgp_comm (comm), m_rrtmgp_params (params) {
+    RRTMGPRadiation::RRTMGPRadiation (const ekat::Comm& comm, const ekat::ParameterList& params) 
+        : AtmosphereProcess::AtmosphereProcess(), m_rrtmgp_comm (comm), m_rrtmgp_params (params) {
         /*
          * Anything that can be initialized without grid information can be initialized here.
          * I.e., universal constants, options, etc.
          */
+          if (!m_rrtmgp_params.isParameter("Grid")) {
+              m_rrtmgp_params.set("Grid",std::string("SE Physics"));
+          }
+
+          m_initializer = create_field_initializer<RRTMGPInputsInitializer>();
     }  // RRTMGPRadiation::RRTMGPRadiation
 
     void RRTMGPRadiation::set_grids(const std::shared_ptr<const GridsManager> grids_manager) {
 
+        using namespace ekat::units;
+
+        auto kgkg = kg/kg;
+        kgkg.set_string("kg/kg");
+        auto m3 = m * m * m;
+        m3.set_string("m3");
+        auto Wm2 = W / m / m;
+        Wm2.set_string("Wm2");
+        auto nondim = m/m;  // dummy unit for non-dimensional fields
+        auto micron = m / 1000000;
+
+        auto VL  = FieldTag::VerticalLevel;
+        auto COL = FieldTag::Column;
+        auto VAR = FieldTag::Variable;
+        auto CMP = FieldTag::Component;
+        constexpr int NVL = 42;  /* TODO THIS NEEDS TO BE CHANGED TO A CONFIGURABLE */
+
+        auto grid = grids_manager->get_grid("Physics");
+        const int num_dofs = grid->get_num_local_dofs();
+        //const int ncol = 128; //num_dofs;
+        const int ncol = num_dofs;
+
+        int nswbands = 14;
+        int nlwbands = 16;
+        int ngas = 8;
+//        string1d gas_names("gas_names",ngas);
```

Comment:
I assume that these commented lines are here because we want to use them in the future?

Reply:
Nah, here because I wasn't sure where to put them when I was developing. Removed.